### PR TITLE
feat(useElementSize): new function

### PR DESCRIPTION
**Pull request by sxzz**

### CODE REVIEW

**Comment by antfu** [comment]
Maybe `useElementSize`? `useSize` feels a bit too general

**Comment by sxzz** [comment]
OK. Already edited.

File: packages/core/useElementSize/index.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+export interface Size {
+  width: number
+  height: number
+}
+
+/**
+ * Reactive size of an HTML element.
+ *
+ * @see   {@link https://vueuse.js.org/useResizeObserver}
```
Comment (antfu) [inline packages/core/useElementSize/index.ts:13]:
```suggestion
 * @see   {@link https://vueuse.js.org/useElementSize}
```

File: packages/core/useElementSize/index.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
+import { MaybeRef } from '@vueuse/shared'
+import { ref } from 'vue-demi'
+import { ResizeObserverOptions, useResizeObserver } from '../useResizeObserver'
+
+export interface Size {
```
Comment (antfu) [inline packages/core/useElementSize/index.ts:5]:
```suggestion
export interface ElementSize {
```

File: packages/core/useElementSize/index.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+ * Reactive size of an HTML element.
+ *
+ * @see   {@link https://vueuse.js.org/useElementSize}
+ * @param target
+ * @param callback
+ * @param options
+ */
+export function useElementSize(
+  target: MaybeRef<Element | null | undefined>,
+  initialSize: Size = { width: Infinity, height: Infinity },
```
Comment (antfu) [inline packages/core/useElementSize/index.ts:20]:
```suggestion
  initialSize: ElementSize = { width: 0, height: 0 },
```

Making the default size to zero makes more sense to me. (just like an empty <div/>)

**Comment by sxzz** [comment]
Ok thanks, it has been edited.

**Comment by antfu** [comment]
I have rethought it a bit, do you think it's better to make it expose more properties from an element?

Something like 

```js
const { x, y, width, height } = useElementBounding(el)
```

https://developer.mozilla.org/en-US/docs/Web/API/Element/getBoundingClientRect

**Comment by antfu** [comment]
And so that `useMouseInElement` can even reuse it to be better abstracted.

**Comment by sxzz** [comment]
I don’t think it is necessary. This function is just to simplify getting the element size.
See [useSize](https://github.com/streamich/react-use/blob/master/docs/useSize.md)

**Comment by sxzz** [comment]
Maybe can add another `useElementBounding` function

**Comment by antfu** [comment]
Ok fair enough, would you like to add it as well? Thanks

**Review by antfu — APPROVED** [review]

**Comment by sxzz** [comment]
Yes, maybe later.

**Comment by antfu** [comment]
@sxzz Planned to do it now as I am going to refactor useMouseInElement, let me know if you are already working on it in case of duplicated works. Thanks.

**Comment by sxzz** [comment]
No, I haven’t started do it yet, so please~😅

**Pull request merged**